src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeopleTable';
+import { Person } from '../../types';
+import { getPeople } from '../../api';
+import { PeopleFilters } from '../PeopleFilters';
+import { setPeopleWithParents } from '../../utils/setPeopleWithParents';
+import { filterPeople } from '../../utils/filterPeople';
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [searchParams] = useSearchParams();
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query');
+  const centuries = searchParams.getAll('centuries');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  const loadPeople = useCallback(async () => {
+    try {
+      setIsLoading(true);
+      const peopleFromServer = await getPeople();
+
+      const peopleWithParents = setPeopleWithParents(peopleFromServer);
+
+      setPeople(peopleWithParents);
+    } catch {
+      setHasLoadingError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    setHasLoadingError(false);
+    loadPeople();
+  }, []);
+
+  const visiblePeople
+    = filterPeople(people, sex, query, centuries, sort, order);
```

Comment:
```suggestion
  const visiblePeople = (
    filterPeople(people, sex, query, centuries, sort, order)
  );
```

Reply:
will be great to create separate functions for filtering and sorting and move the result of each of them in `useMemo`